include/tsid/contacts/contact-two-frame-positions.hpp

```diff
@@ -0,0 +1,132 @@
+  typedef pinocchio::SE3 SE3;
+
+  ContactTwoFramePositions(const std::string& name, RobotWrapper& robot,
+                           const std::string& frameName1,
+                           const std::string& frameName2,
+                           const double minNormalForce,
+                           const double maxNormalForce);
+
+  /// Return the number of motion constraints
+  virtual unsigned int n_motion() const;
+
+  /// Return the number of force variables
+  virtual unsigned int n_force() const;
+
+  virtual const ConstraintBase& computeMotionTask(const double t,
+                                                  ConstRefVector q,
+                                                  ConstRefVector v, Data& data);
+
+  virtual const ConstraintInequality& computeForceTask(const double t,
+                                                       ConstRefVector q,
+                                                       ConstRefVector v,
+                                                       const Data& data);
+
+  virtual const Matrix& getForceGeneratorMatrix();
+
+  virtual const ConstraintEquality& computeForceRegularizationTask(
+      const double t, ConstRefVector q, ConstRefVector v, const Data& data);
+
+  const TaskSE3Equality& getMotionTask() const;
+  const ConstraintBase& getMotionConstraint() const;
+  const ConstraintInequality& getForceConstraint() const;
+  const ConstraintEquality& getForceRegularizationTask() const;
+  double getMotionTaskWeight() const;
+  const Matrix3x& getContactPoints() const;
+
+  double getNormalForce(ConstRefVector f) const;
+  double getMinNormalForce() const;
+  double getMaxNormalForce() const;
+
+  const Vector&
+  Kp();  // cannot be const because it set a member variable inside
+  const Vector&
+  Kd();  // cannot be const because it set a member variable inside
+  void Kp(ConstRefVector Kp);
+  void Kd(ConstRefVector Kp);
+
+  bool setContactNormal(ConstRefVector contactNormal);
+
+  bool setFrictionCoefficient(const double frictionCoefficient);
+  bool setMinNormalForce(const double minNormalForce);
+  bool setMaxNormalForce(const double maxNormalForce);
+  bool setMotionTaskWeight(const double w);
+  void setReference(const SE3& ref);
+  void setForceReference(ConstRefVector& f_ref);
+  void setRegularizationTaskWeightVector(ConstRefVector& w);
+
+  /**
+   * @brief Specifies if properties of the contact point and motion task
+   * are expressed in the local or local world oriented frame. The contact
+   * forces, contact normal and contact coefficients are interpreted in
+   * the specified frame.
+   *
+   * @param local_frame If true, use the local frame, otherwise use the
+   * local world oriented
+   */
+  void useLocalFrame(bool local_frame);
+
+ protected:
+  void updateForceInequalityConstraints();
+  void updateForceRegularizationTask();
+  void updateForceGeneratorMatrix();
+
+  TaskTwoFramesEquality m_motionTask;
+  TaskSE3Equality m_dummyMotionTask;
```

Comment:
I had not noticed this in my first review, thanks for pointing it out. Indeed using this dummyMotionTask is not ideal. It would probably be better to change the API of ContactBase, so that the method `getMotionTask` returns a `MotionTaskBase` rather than an `SE3EqualityTask`. Would you agree with that?